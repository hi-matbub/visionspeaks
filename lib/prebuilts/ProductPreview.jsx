import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Loading from '../components/Loading';

const ProductPreview = ({ product }) => {
  const [primaryImage, setPrimaryImage] = useState();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (!product) setLoading(true);
    else {
      setPrimaryImage(product.images[0]);
      setLoading(false);
    }
  }, [product]);

  const handlePrimaryImageToggle = (i) => setPrimaryImage(i);

  return (
    <>
      {loading ? (
        <Loading data={product} />
      ) : (
        <section className="d-flex flex-column justify-content-center align-items-center p-2">
          <img
            className="primary_image_preview"
            src={primaryImage}
            alt="view product"
          />
          <div className="img_previews">
            {!product.images.length <= 1
              ? null
              : product.images.map((i) => (
                  <button
                    type="button"
                    key={i}
                    className="img_preview_button"
                    onClick={() => handlePrimaryImageToggle(i)}
                  >
                    <img src={i} alt="alt product views" />
                  </button>
                ))}
          </div>
        </section>
      )}
    </>
  );
};

ProductPreview.propTypes = {
  product: PropTypes.shape({
    images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  }).isRequired,
};

export default ProductPreview;
