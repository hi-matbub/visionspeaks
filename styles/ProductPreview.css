.primary_image_preview {
  height: 250px;
  width: 250px;
}

.img_preview_button {
  margin: 0.25em;
}

.img_preview_button,
.img_preview_button > img {
  height: 100px;
  width: 100px;
  border: 0;
  background-color: initial;
}