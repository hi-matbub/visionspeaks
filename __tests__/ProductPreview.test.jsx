import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ProductPreview from '../lib/prebuilts/ProductPreview';

const product = {
  name: 'product',
  description: 'this is an awesome product you should buy 2',
  price: '$2.87',
  id: 'abc_123',
  images: [
    '/path_01.png',
    '/path_02.png',
    '/path_03.png',
  ],
  qty: 12,
};

it('ProductPreview component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ProductPreview product={product} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('ProductPreview component renders unchanged', () => {
  const tree = renderer.create(<ProductPreview product={product} />).toJSON();
  expect(tree).toMatchSnapshot();
});
