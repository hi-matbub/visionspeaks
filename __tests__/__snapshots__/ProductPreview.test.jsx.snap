// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPreview component renders unchanged 1`] = `
<section
  className="d-flex flex-column justify-content-center align-items-center p-2"
>
  <img
    alt="view product"
    className="primary_image_preview"
  />
  <div
    className="img_previews"
  />
</section>
`;
